src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -0,0 +1,277 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.Abstractions;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Messaging.Pumps.ServiceBus
+{
+    public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessage, AzureServiceBusMessageContext>
+    {
+        private bool _isHostShuttingDown;
+        private MessageReceiver _messageReceiver;
+        private readonly MessageHandlerOptions _messageHandlerOptions;
+        private readonly AzureServiceBusMessagePumpSettings _messagePumpSettings;
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="configuration">Configuration of the application</param>
+        /// <param name="serviceProvider">Collection of services that are configured</param>
+        /// <param name="logger">Logger to write telemetry to</param>
+        protected AzureServiceBusMessagePump(IConfiguration configuration, IServiceProvider serviceProvider,
+            ILogger logger)
+            : base(configuration, serviceProvider, logger)
+        {
+            _messagePumpSettings = serviceProvider.GetRequiredService<AzureServiceBusMessagePumpSettings>();
+            _messageHandlerOptions = DetermineMessageHandlerOptions(_messagePumpSettings);
```

Comment:
nitpick. Not sure why this private method (`DetermineMessageHandlerOptions`) is moved all below when it's only used here.

Reply:
Habbit of structure, of some sorts at least:
- ctor / fields / properties
- public methods
- privates

I use F12 for navigation anyway but if it's a problem I can move it

Reply:
If there were multiple methods, I would definitly place more closer to the ctor; but only this one is fine by me.